tests/parser/UDQTests.cpp

```diff
@@ -2932,3 +2926,90 @@ BOOST_AUTO_TEST_CASE(UDQ_Update_SummaryState)
     BOOST_CHECK_CLOSE(st.get_group_var("BB", "GUNDA_ST"), -123.4 , 1.0e-8);
     BOOST_CHECK_CLOSE(st.get_group_var("G1", "GUNDA_ST"),  652.44, 1.0e-8);
 }
+
+BOOST_AUTO_TEST_CASE(UDQ_WITH_UDT_FIELD)
+{
+    std::string valid = R"(
+SCHEDULE
+
+UDT
+ 'TU_FBHP' 1 /
+ 'LC'  100.0  500.0 / -- FOPR values
+       100.0  180.0 / -- FBHP values
+/
+/
+
+UDQ
+ASSIGN FU_FOPR 110.0 /
+ASSIGN FU_WBHP 0 /
+DEFINE FU_WBHP0 FU_WBHP /
+DEFINE FU_WBHP TU_FBHP[FU_FOPR] UMIN FU_WBHP0 /
+/
+
+)";
+
+    auto schedule = make_schedule(valid);
+    UDQState udq_state(0);
+    SummaryState st(TimeService::now());
+    UDQFunctionTable udqft;
+    WellMatcher wm(NameOrder({"W1", "W2", "W3"}));
+    const auto& udq = schedule.getUDQConfig(0);
+    auto segmentMatcherFactory = []() { return std::make_unique<SegmentMatcher>(ScheduleState {}); };
+    UDQContext context(udqft, wm, udq.tables(), segmentMatcherFactory, st, udq_state);
+
+    const auto& ass = udq.assign("FU_WBHP");
+    context.update_assign("FU_WBHP", ass.eval());
+    const auto& ass2 = udq.assign("FU_FOPR");
+    context.update_assign("FU_FOPR", ass2.eval());
+    const auto& def = udq.define("FU_WBHP");
+    auto res = def.eval(context);
+
+    BOOST_CHECK_EQUAL(res.size(), 1U);
+    BOOST_CHECK_EQUAL(res[0].get(), 100.0 + (180.0 - 100.0) * (110.0 - 100.0) / (500.0 - 100.0));
```

Comment:
Pooh, this took quite some time to understand why this 102 and not 100. Did not initially see that we never evaluated `DEFINE FU_WBHP0 FU_WBHP` and hence `FU_WBHP0` has not value.

Is that missing evaluation intentional? Then a comment here would probably help with understanding for dummies like me.

Reply:
just an oversight. it doesn't matter since stuff gets value 0 by default if they are not defined, but i have updated the PR.